service/history/hsm/tree.go

```diff
@@ -452,6 +554,47 @@ func (n *Node) Sync(incomingNode *Node) error {
 
 	// sync LastUpdateVersionedTransition last as MachineTransition can't correctly handle it.
 	n.persistence.LastUpdateVersionedTransition = incomingInternalRepr.LastUpdateVersionedTransition
+
+	if incomingInternalRepr.Deleted {
```

Comment:
This isn't how we'll be replicating tombstones.

Reply:
Will handle replication in Part 2. Removed